protocol/communities/manager.go

```diff
@@ -1455,6 +1482,130 @@ func (m *Manager) IsSeedingHistoryArchiveTorrent(communityID types.HexBytes) boo
 	return exists && ok
 }
 
+func (m *Manager) DownloadHistoryArchivesByMagnetlink(communityID types.HexBytes, magnetlink string) ([]string, error) {
+
+	id := communityID.String()
+	ml, _ := metainfo.ParseMagnetUri(magnetlink)
+
+	log.Println("Adding torrent from magnetlink for community: ", id)
+	torrent, err := m.torrentClient.AddMagnet(magnetlink)
+	if err != nil {
+		return nil, err
+	}
+	m.torrentTasks[id] = ml.InfoHash
+
+	<-torrent.GotInfo()
+	files := torrent.Files()
+
+	indexFile := files[1]
```

Comment:
We always expect exactly two files `data` and `index`. So accessing `[1]` is safe, but we should probably still dynamically determine, in case we'll add more files to these torrent sin the future (maybe due to https://github.com/status-im/specs/pull/167)

Reply:
Can it be that `files` is of different length? (say 1 or 0), even in cases where the torrent is malformed.
If that's the case, best to error check and return an error, since a panic would just crash the app, and if someone manages to push a malformed torrent for example, it would just crash all clients.

Reply:
Yea good point. I'll update it so it'll search for an `index` file and if it doesn't exist it'll return early as the torrent is malformed

Reply:
Updated